File: 3rd Semester/Advanced Programming Methods(Done)/Laboratory/C#LabAssignment/sharplab/Repo/IRepo.cs
using sharplab.Model.Utils;
using System;
using System.Collections.Generic;
using System.Text;

namespace sharplab.Repo
{
    interface IRepo
    {
        PrgState GetCrtPrg();
        void LogPrgStateExec();
        List<PrgState> GetPrgList();
    }
}

File: 3rd Semester/Advanced Programming Methods(Done)/Laboratory/src (BUX)/Repo/IRepo.java
package Repo;

import Model.PrgState;

import java.io.IOException;
import java.util.List;

public interface IRepo {
    void logPrgStateExec(PrgState state);
    void CloseFile();
    List<PrgState> getPrgList();
    void setPrgList(List<PrgState> st);
    void addPrg(PrgState p);
    void serializePrgState(PrgState myState) throws IOException;
    PrgState deserializePrgState() throws IOException, ClassNotFoundException;
}
